test/functional/lua/fs_spec.lua

```diff
@@ -308,26 +309,122 @@ describe('vim.fs', function()
       )
     end)
 
-    it('works with UNC paths', function()
-      eq('//foo', vim.fs.normalize('//foo')) -- UNC path
-      eq('//foo/bar', vim.fs.normalize('//foo//bar////')) -- UNC path
-      eq('/foo', vim.fs.normalize('///foo')) -- Not a UNC path
-      eq('/', vim.fs.normalize('//')) -- Not a UNC path
-      eq('/', vim.fs.normalize('///')) -- Not a UNC path
-      eq('/foo/bar', vim.fs.normalize('/foo//bar////')) -- Not a UNC path
-    end)
-
-    if is_os('win') then
+    if not is_os('win') then
+      it('preserves leading double slashes in POSIX paths', function()
+        eq('//foo', vim.fs.normalize('//foo'))
+        eq('//foo/bar', vim.fs.normalize('//foo//bar////'))
+        eq('/foo', vim.fs.normalize('///foo'))
+        eq('/', vim.fs.normalize('//'))
```

Comment:
I think I wrote this, but I'm not sure this is true in hindsight. I don't see a reason why this should be collapsed.

> A pathname that begins with two successive slashes may be interpreted in an implementation-defined manner, although more than two leading slashes shall be treated as a single slash.

I still interpret `//` to fall under a pathname that begins with two successive slashes.

Reply:
Ah, it's actually easier to not collapse it, so I'm glad to hear that it's not necessary

Reply:
But I could be wrong. I think @justinmk suggested this behavior in the first place, but I am not sure why.

Reply:
Looking at https://stackoverflow.com/questions/20690828/what-does-double-slash-in-cd-mean-in-linux, specifically this comment:

> The // does not mean nothing. It's defined by POSIX such that it may be interpreted in an implementation-defined manner, which means the system is free to do with it what it will. For example, on Cygwin, cd // followed by ls eventually returns ls: reading directory '.': Permission denied. whereas cd / followed by ls returns a listing of the root directory. (Cygwin uses a root of // to identify //hostname/path.)

Seems like `//` by itself is special as well